enhancements/bare-metal-fulfillment/README.md

```diff
@@ -0,0 +1,267 @@
+Afterwards, the tenant will be able to perform selected operations on the host: inventory information, power control, and
+serial console access. Any further operations - for example, image-based provisioning - are outside the scope of the O-SAC
+solution and must be integrated by the cloud provider.
+
+In this context, a “tenant” is a person with admin rights to the requested bare metal resources; at the MOC, this person would be
+a ColdFront PI or manager.
+
+## Motivation
+
+Bare metal fulfillment is a fundamental requirement for O-SAC, identified as an explicit need for the MOC. It will also serve
+as the basis of bare metal configuration for other fulfillment workflows, such as OpenShift cluster fulfillment.
+
+### User Stories
+
+* As a tenant, I want to be able to see available bare metal resource classes.
+* As a tenant, I want to be able to select hosts for a bare metal cluster by resource class.
+* As a tenant, I want to be able to view inventory details for my hosts.
+* As a tenant, I want to be able to connect to the serial console of my hosts.
+* As a tenant, I want to be able to perform basic power control (power on/power off/reset) of my hosts.
+* As a tenant, I want to be able to attach hosts to existing networks.
+* As a tenant, I want to be able to modify the network connectivity of my hosts by attaching or detaching networks.
+* As a tenant, I want to be able to manage connectivity at the interface level in order to support hardware that may have both regular and high performance interfaces (for example, some hosts may have high-bandwidth connectivity for GPU workloads, and it may be necessary to attach specific networks to these interfaces).
+
+### Goals
+
+We will implement bare metal fulfillment in the O-SAC solution, covering the user stories described above. This implementation will
+be first deployed and tested in the MOC; as a result, the implementation will rely upon ESI for bare metal management (although
+it will be designed to allow easy replacement of the bare metal management layer).
+
+### Non-Goals
+
+We will not look into alternatives for ESI; that work will be covered through a future enhancement.
+
+We will not substantially modify the existing cluster fulfillment workflow, although we believe bare metal fulfillment should
+eventually be used as the basis of bare metal management within cluster fulfillment.
+
+## Proposal
+
+The implementation of bare metal fulfillment relies on three concepts:
+
+* HostPool: A collection of requested bare metal machines, along with their desired network configuration.
+* Host: An ephemeral resource created when bare metal resources are acquired from the underlying bare metal management layer. They represent the assignment of a bare metal resource to a HostPool, and provide an endpoint for host-specific operations.
+* HostClass: Defines a set of bare metal properties, such as memory, CPU, and network interfaces. Each Host has a single HostClass attribute.
```

Comment:
should we say specified by the provider?

Reply:
That's fair - added!